src/main/scala/stuff/monads.scala

```diff
@@ -0,0 +1,101 @@
+package ohnosequences.stuff
+
+import naturalTransformations._
+import products._
+import functions._
+
+abstract
+class Monad {
+
+  type On <: Functor.endo
+  val on   : Functor.is[On]
+
+  // TODO do I need another Functor.is here?
```

Comment:
Where do we need that?

Reply:
This is about whether we need the more precise type

``` scala
val μ: Functor.is[Functor.Composition[Functor.is[On],Functor.is[On]]] ~> Functor.is[On]
```

We *could*, I'm not sure on this one.